bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
Can you give an estimate to go the `bokeh-jqui` route? To make sure I understand, there would be some notebook issues with this route? Or we'd keep prefixes for the few things that need them? If this is going to take more than a day or so we'd need to discuss prioritization.